### PR TITLE
Preferences: add translation feature

### PR DESCRIPTION
**Pull request by cristina-suteu**
Scopy translation feature
auto option for setting app to os language, if available
browse option for custom translation files
resources/languages directory for bundled translations

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -111,19 +112,34 @@ int main(int argc, char **argv)
...
+    const QStringList content = in.readAll().split("\n");
 	QString language="default";
-	for(auto s : content)
-	{
-		if(s.startsWith("language"))
-		{
-			language=s.split("=")[1];
+	for(auto s : content) {
+		if(s.startsWith("language")) {
+			language=s.split("=")[1].trimmed();
```
Comment (DanielGuramulta) [inline src/main.cpp:119]: From what I understand in this lines, the ini file, `Preferences.ini`, is manually opened and parsed. I think the language should be retrieved through the Preferences API.

File: src/main.cpp [hunk 2]
```diff
@@ -111,19 +112,34 @@ int main(int argc, char **argv)
...
+	else if(languages.contains(language+".qm")) {
+		languageFileName=QDir(QCoreApplication::applicationDirPath()+"/resources/languages/"+language+".qm").path();
+	}
+	else { 
+		languageFileName = language;
+	}
 
-	QString languageFileName=QDir(QCoreApplication::applicationDirPath()+"/resources/languages/"+language+".qm").path();
+	f.close();
 	myappTranslator.load(languageFileName);
```
Comment (DanielGuramulta) [inline src/main.cpp:143]: Wouldn't it be easier to just
```cpp
        if (translator->load(locale, QCoreApplication::applicationName(), "_", dir)) {
            app.installTranslator(translator);
        }
```
where dir would be the directory of the languages. In this way the line above will try to load `scopy_en` or `scopy_de`, any scopy_<language> depending on the language given by the locale. If no language is found in this directory then no translator should be installed.
The languages (*.qm files) should be compiled in a resource and attached to the executable. The current implementation won't work for flatpak and the paths to the translation files would need to be given manually.

File: src/main.cpp [hunk 3]
```diff
@@ -173,3 +189,4 @@ int main(int argc, char **argv)
 	}
 	return app.exec();
 }
+
```
Comment (DanielGuramulta) [inline src/main.cpp:192]: nitpik: extra line

File: src/preferences.cpp [hunk 4]
```diff
@@ -152,24 +156,68 @@ Preferences::Preferences(QWidget *parent) :
...
 	pref_api->load(settings);
-
 	connect(ui->languageCombo, &QComboBox::currentTextChanged, [=](QString lang) {
-		language = lang;
-		if (m_initialized)
-			ui->label_restart->setVisible(true);
-		else
-			m_initialized = true;
-		Q_EMIT notify();
+			if(lang=="browse"){
```
Comment (DanielGuramulta) [inline src/preferences.cpp:167]: Should format this code:
```cpp
if () {
    ...
} else {
    ...
}
```

File: src/preferences.cpp [hunk 5]
```diff
@@ -152,24 +156,68 @@ Preferences::Preferences(QWidget *parent) :
...
+				if (m_initialized)
+					ui->label_restart->setVisible(true);
+				else
+					m_initialized = true;
+        			}
+			Q_EMIT notify();
 	});
 }
 
+QStringList Preferences::getOptionsList()
```
Comment (DanielGuramulta) [inline src/preferences.cpp:198]: This also needs formatting

File: src/preferences.cpp [hunk 6]
```diff
@@ -198,6 +246,7 @@ void Preferences::notifyChange()
 
 void Preferences::showEvent(QShowEvent *event)
 {
+
```
Comment (DanielGuramulta) [inline src/preferences.cpp:249]: extra line

File: src/preferences.cpp [hunk 7]
```diff
@@ -152,24 +156,68 @@ Preferences::Preferences(QWidget *parent) :
...
+QStringList Preferences::getOptionsList()
+{   
+	QFile f(getPreferenceIniFile());
+	f.open(QFile::ReadOnly);
+    QTextStream in (&f);
+	const QStringList content = in.readAll().split("\n");
+	QString language;
+	for(auto s : content) {
+		if(s.startsWith("language")){
+			language=s.split("=")[1].trimmed();
```
Comment (DanielGuramulta) [inline src/preferences.cpp:207]: I think there already is a property called language for this that uses `getLanguage`, `setLanguage`, is the parsing of the ini file needed?

**Comment by adisuciu** [comment]
> This pull request seems to add a little more than just `add translation feature`. I see there are also some changes regarding CI, and Signal Generators `MATLAB` files behavior. I don't think this changes should be together.



> This pull request seems to add a little more than just `add translation feature`. I see there are also some changes regarding CI, and Signal Generators `MATLAB` files behavior. I don't think this changes should be together.

Those changes are already on master. PR just needs a rebase.

**Pull request merged**